pkg/operator/certrotation/cabundle.go

```diff
@@ -88,12 +87,16 @@ func (c CABundleConfigMap) EnsureConfigMapCABundle(ctx context.Context, signingC
 	}
 
 	if modified {
-
-		actualCABundleConfigMap, updated, err := resourceapply.ApplyConfigMap(ctx, c.Client, c.EventRecorder, caBundleConfigMap)
+		var actualCABundleConfigMap *corev1.ConfigMap
+		if len(caBundleConfigMap.ResourceVersion) == 0 {
+			actualCABundleConfigMap, err = c.Client.ConfigMaps(c.Namespace).Create(ctx, caBundleConfigMap, metav1.CreateOptions{})
```

Comment:
Given that we would like to report successful creation as well why not change the code to be more explicit.
For example:
1. we would introduce a new `var creationRequired bool` or `var creationNeeded bool` variable.
2. we would set the new variable at[ line 60](https://github.com/openshift/library-go/blob/d3df6775ec935973df3fdc6c231147865e4bad6a/pkg/operator/certrotation/cabundle.go#L60) 
3. we would rename `modified` to `updateRequired` or `updateNeeded`
4. we would introduce a branch before line 89, for example:
```
if creationRequired {
 create
 report
 assign to `caBundleConfigMap`
 } else if updateRequired {
 update 
 report
 assign to `caBundleConfigMap`
 }
 ```

Reply:
Alternatively we could move the creation to [line 53](https://github.com/openshift/library-go/blob/d3df6775ec935973df3fdc6c231147865e4bad6a/pkg/operator/certrotation/cabundle.go#L53).

The creation is a rare operation and immediately after creation the sync loop will be triggered and we can issue an update.

Reply:
Good idea, now all functions should have `exists` and `modified`

Reply:
we don't want to report https://github.com/openshift/library-go/blob/master/pkg/operator/resource/resourceapply/event_helpers.go#L25 anymore ?

Reply:
Fixed